addons/web/static/src/views/widgets/ribbon/ribbon.js

```diff
@@ -0,0 +1,51 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { standardViewWidgetProps } from "../standard_view_widget_props";
+
+const { Component } = owl;
+
+/**
+ * This widget adds a ribbon on the top right side of the form
+ *
+ *      - You can specify the text with the title attribute.
+ *      - You can specify the tooltip with the tooltip attribute.
+ *      - You can specify a background color for the ribbon with the bg_color attribute
+ *        using bootstrap classes :
+ *        (bg-primary, bg-secondary, bg-success, bg-danger, bg-warning, bg-info,
+ *        bg-light, bg-dark, bg-white)
+ *
+ *        If you don't specify the bg_color attribute the bg-success class will be used
+ *        by default.
+ */
+class RibbonWidget extends Component {
+    setup() {
+        this.className = this.props.className;
+        if (this.props.text.length > 15) {
+            this.className += " o_small";
+        } else if (this.props.text.length > 10) {
+            this.className += " o_medium";
+        }
+    }
```

Comment:
I would do that in a getter, in case className or text would change during the life of the component